core/dbt/clients/registry.py

```diff
@@ -19,26 +16,10 @@ def _get_url(url, registry_base_url=None):
     return '{}{}'.format(registry_base_url, url)
 
 
-def _wrap_exceptions(fn):
-    @wraps(fn)
-    def wrapper(*args, **kwargs):
-        max_attempts = 5
-        attempt = 0
-        while True:
-            attempt += 1
-            try:
-                return fn(*args, **kwargs)
-            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as exc:
-                if attempt < max_attempts:
-                    time.sleep(1)
-                    continue
-                raise RegistryException(
-                    'Unable to connect to registry hub'
-                ) from exc
-    return wrapper
-
-
-@_wrap_exceptions
+def _get_with_retries(path, registry_base_url=None):
+    return connection_exception_retry(lambda: _get(path, registry_base_url), 5)
```

Comment:
Not a required change, but I _think_ this might be a good place to use `functools.partial`!(https://docs.python.org/3/library/functools.html#functools.partial)

```suggestion
    get_fn = partial(_get, path, registry_base_url)
    return connection_exception_retry(get_fn, 5)
```